test/index.spec.ts

```diff
@@ -0,0 +1,208 @@
+import { unixfs, type UnixFS } from '@helia/unixfs'
+import { Configuration, RemotePinningServiceClient, Status } from '@ipfs-shipyard/pinning-service-client'
+import { expect } from 'aegir/chai'
+import { createHelia } from 'helia'
+import sinon, { type SinonSandbox, type SinonStub } from 'sinon'
+import { FailedToConnectToDelegates } from '../src/errors.js'
+import { type HeliaRemotePinner, createRemotePinner } from '../src/index.js'
+import type { Helia } from '@helia/interface'
+
+const encoder = new TextEncoder()
+
+describe('@helia/remote-pinning', function () {
+  let sinonSandbox: SinonSandbox
```

Comment:
You can use [sinon-ts](https://www.npmjs.com/package/sinon-ts) to stub interfaces and objects with types - it  will simplify things a little and you won't need to specify pinning service details to run the tests.

Reply:
since the pinning-service we use for testing is already a mocked up library (`mock-ipfs-pinning-service`), i think the setup is fine, and something we've established a pattern for elsewhere. i'd rather not stub/mock up all the things myself.